recipes-sota/aktualizr/aktualizr-device-prov-creds.bb

```diff
@@ -32,9 +32,10 @@ export SOTA_CAKEY_PATH
 do_install() {
     if [ -n "${SOTA_PACKED_CREDENTIALS}" ]; then
         if [ -z ${SOTA_CACERT_PATH} ]; then
-            SOTA_CACERT_PATH=${DEPLOY_DIR_IMAGE}/CA/cacert.pem
-            SOTA_CAKEY_PATH=${DEPLOY_DIR_IMAGE}/CA/ca.private.pem
-            mkdir -p ${DEPLOY_DIR_IMAGE}/CA
+            SOTA_CA_DIR=${DEPLOY_DIR}/sota-ca
```

Comment:
It's quite unusual to store things in `${DEPLOY_DIR}` directly.
Making this recipe MACHINE_ARCH sounds better to me.

Reply:
> It's quite unusual to store things in `${DEPLOY_DIR}` directly.
> Making this recipe MACHINE_ARCH sounds better to me.

But the CA files we generate are certainly not machine-specific. They aren't specific to anything, really. It's an unnecessary dependency that we can easily avoid. Is there a better place to put them?

Reply:
Well I think the wrong idea we had is that the MACHINE_ARCH packages means that the files should be generated differently for each machine. It seems like it's more about that the resulting files are not shared between machines.

And it's different from the aktualizr case where the recipe is actually expensive to build, this one is very cheap. I'm not convinced it's worth the optimization

But this code path also generates a lot of warnings, it's a bit of a pathological case.

Reply:
> this one is very cheap. I'm not convinced it's worth the optimization

That is a fair point. Also, this recipe is fairly unrealistic in that if you actually use the full extent of it, it puts device-specific credentials in an image that can be shared among many devices, so maybe we should change what we're doing here...

> But this code path also generates a lot of warnings, it's a bit of a pathological case.

What other warnings are you referring to?

Reply:
I was talking about all the bbwarn we added.

Reply:
I see. Should we just remove this whole recipe? It's doing the "wrong" thing as mentioned above, it's not covered by oe-selftest, and AFAIK it isn't used by anyone. If you want device credential provisioning, you can't really use bitbake to generate device-specific credentials.

Reply:
Yes maybe it doesn't make much sense. A middle ground could be to just remove the logic in `if [ -z ${SOTA_CACERT_PATH} ];` and just bbfail there. It would solve the machine arch dependency issue that we're trying to solve here and we can still remove the recipe later.

Reply:
Decent suggestion, but I think it's time to just remove it and be done with it, so that's what I did.

Reply:
> It's quite unusual to store things in `${DEPLOY_DIR}` directly.
> Making this recipe MACHINE_ARCH sounds better to me.

It's wrong to install anything to any DEPLOY* variable directly.

do_install should install just to D and do_deploy installs files from D to DEPLOYDIR and then creates deploy sstate which installs it from DEPLOYDIR to DEPLOY_DIR_IMAGE.

Writting to DEPLOY_DIR directly from do_install (or any other global directory from any other task not covered by sstate) only means that the files won't be there once the component is reused from sstate.

E.g. if you build aktualizr-device-prov-creds, then the files will be in deploy, but if you delete tmp-glibc and build it again, then the files will be missing (because do_install wasn't executed at all when there is valid sstate-cache for aktualizr-device-prov-creds).

So good that this is removed, but keep this in mind for other recipes as well.

Reply:
@shr-project Thanks for the detailed description here! I was starting to put these pieces together and realized that this recipe was just not doing the right thing. It was meant to just be for demonstration anyway, but it was demonstrating it wrongly enough that it was better to just remove it and be done with it.